src/index.ts

```diff
@@ -2249,6 +2256,187 @@ async function claimAllUnclaimedAsSenderPerChain({
 	return txHashes
 }
 
+// Returns args to be passed to makeDepositWithAuthorization function
+// and a EIP-712 message to be signed
+async function prepareGaslessDeposit({
+	address,
+	contractVersion,
+	linkDetails,
+	password,
+}: interfaces.IPrepareGaslessDepositParams): Promise<interfaces.IPrepareGaslessDepositResponse> {
+	if (!PeanutsWithEIP3009.includes(contractVersion)) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: this Peanut version does not support gasless deposits'
+		)
+	}
+
+	if (linkDetails.tokenType !== interfaces.EPeanutLinkType.erc20) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: only erc20 tokens are currently supported for gasless deposits'
+		)
+	}
+
+	if (!linkDetails.tokenAddress || !linkDetails.tokenDecimals) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: token address and decimals must be provided'
+		)
+	}
+
```

Comment:
tokenDecimals should be optional

Check this code out:

// If token details not found in TOKEN_DETAILS, fetch them from the contract
		if (!tokenDetails) {
			try {
				const contractERC20 = new ethers.Contract(tokenAddress, ERC20_ABI, provider)
				const [fetchedSymbol, fetchedName, fetchedDecimals] = await Promise.all([
					contractERC20.symbol(),
					contractERC20.name(),
					contractERC20.decimals(),
				])
				symbol = fetchedSymbol
				name = fetchedName
				tokenDecimals = fetchedDecimals
				tokenAmount = ethers.utils.formatUnits(deposit.amount, tokenDecimals)
			} catch (error) {
				console.error('Error fetching ERC20 info:', error)
			}
		} else {
			symbol = tokenDetails.symbol
			name = tokenDetails.name
			tokenDecimals = tokenDetails.decimals
			tokenAmount = ethers.utils.formatUnits(deposit.amount, tokenDecimals)
		}

Reply:
Makes sense! The code you linked though is from `getLinkDetails`, which works on top of an already existing link, so I did a slightly different change and added a default of 18 decimals, so that it is compliant with the logic in `createLink` (the logic there is:
```ts
if (linkDetails.tokenType == interfaces.EPeanutLinkType.erc1155) {
		linkDetails.tokenDecimals = linkDetails.tokenDecimals ?? 0
	} else {
		linkDetails.tokenDecimals = linkDetails.tokenDecimals ?? 18
	}
```
)